lib/elixir/src/elixir_with.erl

```diff
@@ -57,3 +78,21 @@ build_cases([], DoExpr) ->
   DoExpr;
 build_cases([{_, Meta, _} = Expr | Rest], DoExpr) ->
   {'__block__', Meta, [Expr, build_cases(Rest, DoExpr)]}.
+
+build_else(Meta, WithCases, ElseClauses) ->
+  Result = {'result', Meta, nil},
+  Clauses = [
+    {'->', Meta, [[{{'^', Meta, [?ref(Meta)]}, Result}], Result]}
+    | ElseClauses
+  ] ++ [build_raise(Meta)],
+  {'case', Meta, [WithCases, [{do, Clauses}]]}.
+
+reference(Meta) ->
```

Comment:
Maybe it could be cleaner defining `%WithResult{}` instead of `make_ref`?
@josevalim any preference?